bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -110,67 +110,63 @@ type dmOperationEntry struct {
 	action string
 }
 
-func getDefaultArgs() GcpArgs {
-	return GcpArgs{
-		StorageOption: configtypes.StorageOption{
-			CreatePipelinePersistentStorage: true,
-		},
+// GetPlatform returns the gcp kfapp. It's called by coordinator.GetPlatform
+func GetPlatform(kfdef *kfdefs.KfDef) (kftypes.Platform, error) {
+	_gcp := &Gcp{
+		kfDef:            kfdef,
+		gcpAccountGetter: getGcloudDefaultAccount,
 	}
+	return _gcp, nil
 }
 
-// GetPlatform returns the gcp kfapp. It's called by coordinator.GetPlatform
-func GetPlatform(kfdef *kfdefs.KfDef, platformArgs []byte) (kftypes.Platform, error) {
-	var gcpArgs GcpArgs
-	if platformArgs == nil {
-		gcpArgs = getDefaultArgs()
-	} else {
-		err := json.Unmarshal(platformArgs, &gcpArgs)
-		if err != nil {
-			return nil, err
-		}
+// GetPluginSpec gets the plugin spec.
+func (gcp *Gcp) GetPluginSpec() *GcpPluginSpec {
+	// Not passing a pointer interface is a common cause of deserialization problems
+	pluginSpec := &GcpPluginSpec{}
+
+	// It should be fine to swallow the error
```

Comment:
On second thought we shouldn't swallow the error. I was thinking the only way we would fail is if the plugin wasn't defined in which case returning an empty spec was ok.

But we could also have a case where we have marshaling error; e.g. because of a YAML error or because the spec changed.